src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunILLink.cs

```diff
@@ -441,8 +453,35 @@ private void EnableNonFrameworkTrimming(XDocument project)
                                                  new XElement("action"))));
         }
 
-        private TestProject CreateTestProjectForILLinkTesting(string targetFramework, string mainProjectName, string referenceProjectName, bool usePackageReference = true)
+        private TestProject CreateTestProjectForILLinkTesting(
+            string targetFramework,
+            string mainProjectName,
+            string referenceProjectName = null,
+            bool usePackageReference = true,
+            [CallerMemberName] string callingMethod = "")
```

Comment:
If you are calling this method from a `Theory`, then the different instances will still collide on the path.  The pattern we generally use for this is to also pass in an optional `identifier` parameter, which can be a concatenation of the `Theory` parameters.

Reply:
Thanks!